test/helpers/vertical_nav_helper_test.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class VerticalNavHelperTest < ActionView::TestCase
+
+  test '#backend_api_nav_sections' do
+    @backend_api = FactoryBot.create(:backend_api)
+
+    # if permitted
+    stubs(can?: true)
+    assert_equal(backend_api_nav_sections.pluck(:title), ["Overview", "Analytics", "Methods & Metrics", "Mapping Rules"])
+
+    # if not permitted
+    stubs(can?: false)
+    assert_equal(backend_api_nav_sections.pluck(:title), ["Overview", "Methods & Metrics", "Mapping Rules"])
+
+    # When backend_api is not persisted
+    @backend_api = BackendApi.new
+    assert_equal(backend_api_nav_sections.pluck(:title), [])
```

Comment:
```suggestion
    @backend_api = BackendApi.new
    assert_equal([], backend_api_nav_sections.pluck(:title))
    
    # What about this test?
    @backend_api = nil
    assert_equal([], backend_api_nav_sections)
```
Please fix above too, we always prefer expected vs actual in test unit
```ruby
assert_equal expected, actual
``
![image](https://user-images.githubusercontent.com/64276/142437407-54964f34-c592-4e78-96c1-2cf9b73ed8d6.png)

FYI, in RSpec we do the inverse:

```ruby
expect(actual).to eq expected
``

Reply:
Changes done